src/pymor/reductors/h2.py

```diff
@@ -644,6 +644,201 @@ def reconstruct(self, u):
         )
 
 
+class GapIRKAReductor(GenericIRKAReductor):
+    """Gap-IRKA reductor.
+
+    Parameters
+    ----------
+    fom
+        The full-order |LTIModel| to reduce.
+    mu
+        |Parameter|.
+    """
+    def __init__(self, fom, mu=None):
+        assert isinstance(fom, LTIModel)
+        super().__init__(fom, mu=mu)
+
+    def reduce(self, rom0_params, tol=1e-4, maxit=100, num_prev=1, conv_crit='htwogap',
+               projection='orth'):
+        r"""Reduce using gap-IRKA.
+
+        See [BBG19]_ Algorithm 1.
+
+        Parameters
+        ----------
+        rom0_params
+            Can be:
+
+            - order of the reduced model (a positive integer),
+            - initial interpolation points (a 1D |NumPy array|),
+            - dict with `'sigma'`, `'b'`, `'c'` as keys mapping to
+              initial interpolation points (a 1D |NumPy array|), right
+              tangential directions (|VectorArray| from
+              `fom.input_space`), and left tangential directions
+              (|VectorArray| from `fom.output_space`), all of the same
+              length (the order of the reduced model),
+            - initial reduced-order model (|LTIModel|).
+
+            If the order of reduced model is given, initial
+            interpolation data is generated randomly.
+        tol
+            Tolerance for the convergence criterion.
+        maxit
+            Maximum number of iterations.
+        num_prev
+            Number of previous iterations to compare the current
+            iteration to. A larger number can avoid occasional cyclic
+            behavior.
+        conv_crit
+            Convergence criterion:
+
+            - `'sigma'`: relative change in interpolation points
+            - `'htwogap'`: :math:`\mathcal{H}_2-gap` distance of
+              reduced-order models divided by :math:`\mathcal{L}_2`
+              norm of new reduced-order model
+            - `'ltwo'`: relative :math:`\mathcal{L}_2` distance of
+              reduced-order models
+        projection
+            Projection method:
+
+            - `'orth'`: projection matrix is orthogonalized with respect
+              to the Euclidean inner product,
+            - `'Eorth'`: projection matrix is orthogonalized with
+              respect to the E product.
```

Comment:
It looks like this is copied from `OneSidedIRKAReductor`, but it should be more similar to `IRKAReductor` since there are two projection matrices (in particular, `'biorth'` should be used instead of `'Eorth'`).